src/Interpreter/Interpreter.cpp

```diff
@@ -396,28 +398,34 @@ ExecutionResult Interpreter::visitSwitchInst(llvm::SwitchInst& inst) {
 
   return forks;
 }
-ExecutionResult Interpreter::visitCallInst(llvm::CallInst& call) {
-  auto func = call.getCalledFunction();
+ExecutionResult Interpreter::visitCallBase(llvm::CallBase& callBase) {
+  auto func = callBase.getCalledFunction();
   if (!func)
-    return visitIndirectCall(call);
+    return visitIndirectCall(callBase);
 
   CAFFEINE_ASSERT(!func->isIntrinsic(),
                   fmt::format("bad function call '{}': intrinsic function "
                               "calls should be handled by visitIntrinsic",
                               func->getName().str()));
 
   if (func->empty())
-    return visitExternFunc(call);
+    return visitExternFunc(callBase);
 
```

Comment:
Ok so my thought for this was something like this here
```suggestion
  if (func->empty()) {
    auto prev_bb = ctx->stack_top().current_block;
    auto res = visitExternFunc(callBase);

    auto invoke = llvm::dyn_cast<llvm::InvokeInst>(&callBase);
    if (!invoke)
      return res;

    if (res.empty()) {
      if (ctx->stack_top().current_block == prev_bb)
        performInvokeReturn(*ctx, invoke);
    } else {
      for (auto& ctx : res.contexts()) {
        if (ctx.stack_top().current_block == prev_bb)
          performInvokeReturn(ctx, invoke);
      }
    }
```
This way, we don't have to call `performInvokeReturn` for _every single external function_ and it will avoid doing things when the builtin function does bizarre things with control flow.